evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -400,6 +420,29 @@ def test_predict(mock_predict, mock_fit, example_graph, X_y_binary):
     assert mock_fit.call_count == 3  # Only called during fit, not predict
 
 
+@patch('evalml.pipelines.components.Estimator.fit')
+@patch('evalml.pipelines.components.Estimator.predict')
+def test_predict_repeat_estimator(mock_predict, mock_fit, X_y_binary):
```

Comment:
Is there a test that checks that the data from the components flows through the graph correctly? Like an estimator will call `predict` on the expected data coming from its parents.

Reply:
Yep, there's a mocked test `test_component_graph_evaluation_plumbing` from my previous PR. I realized I never added a test for the `ComponentGraph.input_feature_names` which also reflects the flow of data, so I've added a separate test for that as well!